### ISSUE TITLE
[G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112

### ISSUE THREAD

**Issue by jayanth100798** [unitreerobotics/unitree_IL_lerobot#68]
hello when i run the inference on sim i get this error

(unitree_lerobot) ras_desktop@ras-desktop-X870-AORUS-ELITE-WIFI7:~/unitree_IL_lerobot$ python unitree_lerobot/eval_robot/eval_g1_sim.py     --policy.path=unitree_lerobot/lerobot/outputs/train/2026-02-18/01-24-51_act/checkpoints/020000/pretrained_model     --repo_id=unitreerobotics/G1_Dex3_ToastedBread_Dataset     --arm="G1_29"     --ee="dex3"     --visualization=true 
INFO 2026-02-19 14:34:50 l_g1_sim.py:232 {'arm': 'G1_29',
 'ee': 'dex3',
 'episodes': 0,
 'frequency': 30.0,
 'headless': False,
 'max_episodes': 1200,
 'motion': False,
 'policy': {'chunk_size': 100,
            'device': 'cuda',
            'dim_feedforward': 3200,
            'dim_model': 512,
            'dropout': 0.1,
            'feedforward_activation': 'relu',
            'input_features': {'observation.images.cam_left_high': {'shape': (3,
                                                                              480,
                                                                              640),
                                                                    'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.images.cam_left_wrist': {'shape': (3,
                                                                               480,
                                                                               640),
                                                                     'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.images.cam_right_high': {'shape': (3,
                                                                               480,
                                                                               640),
                                                                     'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.images.cam_right_wrist': {'shape': (3,
                                                                                480,
                                                                                640),
                                                                      'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.state': {'shape': (28,),
                                                     'type': <FeatureType.STATE: 'STATE'>}},
            'kl_weight': 10.0,
            'latent_dim': 32,
            'license': None,
            'n_action_steps': 100,
            'n_decoder_layers': 1,
            'n_encoder_layers': 4,
            'n_heads': 8,
            'n_obs_steps': 1,
            'n_vae_encoder_layers': 4,
            'normalization_mapping': {'ACTION': <NormalizationMode.MEAN_STD: 'MEAN_STD'>,
                                      'STATE': <NormalizationMode.MEAN_STD: 'MEAN_STD'>,
                                      'VISUAL': <NormalizationMode.MEAN_STD: 'MEAN_STD'>},
            'optimizer_lr': 1e-05,
            'optimizer_lr_backbone': 1e-05,
            'optimizer_weight_decay': 0.0001,
            'output_features': {'action': {'shape': (28,),
                                           'type': <FeatureType.ACTION: 'ACTION'>}},
            'pre_norm': False,
            'pretrained_backbone_weights': 'ResNet18_Weights.IMAGENET1K_V1',
            'pretrained_path': 'unitree_lerobot/lerobot/outputs/train/2026-02-18/01-24-51_act/checkpoints/020000/pretrained_model',
            'private': None,
            'push_to_hub': False,
            'replace_final_stride_with_dilation': 0,
            'repo_id': None,
            'tags': None,
            'temporal_ensemble_coeff': None,
            'use_amp': False,
            'use_vae': True,
            'vision_backbone': 'resnet18'},
 'rename_map': {},
 'repo_id': 'unitreerobotics/G1_Dex3_ToastedBread_Dataset',
 'root': '',
 'save_data': False,
 'send_real_robot': False,
 'sim': True,
 'task_dir': './data',
 'use_dataset': False,
 'visualization': True}
INFO 2026-02-19 14:34:50 l_g1_sim.py:240 Making policy.
Loading weights from local directory
14:34:51.161648 INFO     Arguments:                                                eval_g1_sim.py:66
                         EvalRealConfig(repo_id='unitreerobotics/G1_Dex3_ToastedBr                  
                         ead_Dataset', policy=ACTConfig(n_obs_steps=1,                              
                         input_features={'observation.state':                                       
                         PolicyFeature(type=<FeatureType.STATE: 'STATE'>,                           
                         shape=(28,)), 'observation.images.cam_left_high':                          
                         PolicyFeature(type=<FeatureType.VISUAL: 'VISUAL'>,                         
                         shape=(3, 480, 640)),                                                      
                         'observation.images.cam_right_high':                                       
                         PolicyFeature(type=<FeatureType.VISUAL: 'VISUAL'>,                         
                         shape=(3, 480, 640)),                                                      
                         'observation.images.cam_left_wrist':                                       
                         PolicyFeature(type=<FeatureType.VISUAL: 'VISUAL'>,                         
                         shape=(3, 480, 640)),                                                      
                         'observation.images.cam_right_wrist':                                      
                         PolicyFeature(type=<FeatureType.VISUAL: 'VISUAL'>,                         
                         shape=(3, 480, 640))}, output_features={'action':                          
                         PolicyFeature(type=<FeatureType.ACTION: 'ACTION'>,                         
                         shape=(28,))}, device='cuda', use_amp=False,                               
                         push_to_hub=False, repo_id=None, private=None, tags=None,                  
                         license=None,                                                              
                         pretrained_path='unitree_lerobot/lerobot/outputs/train/20                  
                         26-02-18/01-24-51_act/checkpoints/020000/pretrained_model                  
                         ', chunk_size=100, n_action_steps=100,                                     
                         normalization_mapping={'VISUAL':                                           
                         <NormalizationMode.MEAN_STD: 'MEAN_STD'>, 'STATE':                         
                         <NormalizationMode.MEAN_STD: 'MEAN_STD'>, 'ACTION':                        
                         <NormalizationMode.MEAN_STD: 'MEAN_STD'>},                                 
                         vision_backbone='resnet18',                                                
                         pretrained_backbone_weights='ResNet18_Weights.IMAGENET1K_                  
                         V1', replace_final_stride_with_dilation=0,                                 
                         pre_norm=False, dim_model=512, n_heads=8,                                  
                         dim_feedforward=3200, feedforward_activation='relu',                       
                         n_encoder_layers=4, n_decoder_layers=1, use_vae=True,                      
                         latent_dim=32, n_vae_encoder_layers=4,                                     
                         temporal_ensemble_coeff=None, dropout=0.1,                                 
                         kl_weight=10.0, optimizer_lr=1e-05,                                        
                         optimizer_weight_decay=0.0001,                                             
                         optimizer_lr_backbone=1e-05), root='', episodes=0,                         
                         frequency=30.0, arm='G1_29', ee='dex3', motion=False,                      
                         headless=False, sim=True, visualization=True,                              
                         send_real_robot=False, use_dataset=False,                                  
                         save_data=False, task_dir='./data', max_episodes=1200,                     
                         rename_map={})                                                             
[2026-02-19T09:04:51Z INFO  re_grpc_server] Listening for gRPC connections on 0.0.0.0:9876. Connect by running `rerun --connect rerun+http://127.0.0.1:9876/proxy`
[2026-02-19T09:04:51Z INFO  winit::platform_impl::linux::x11::window] Guessed window scale factor: 1
[2026-02-19T09:04:51Z WARN  wgpu_hal::vulkan::instance] Unable to find extension: VK_EXT_physical_device_drm
[2026-02-19T09:04:51Z WARN  wgpu_hal::gles::egl] No config found!
[2026-02-19T09:04:51Z WARN  wgpu_hal::gles::egl] EGL says it can present to the window but not natively

Image client has started, waiting to receive data...
[2026-02-19T09:04:51Z WARN  wgpu_hal::gles::adapter] Max vertex attribute stride unknown. Assuming it is 2048
[2026-02-19T09:04:51Z WARN  wgpu_hal::gles::adapter] Max vertex attribute stride unknown. Assuming it is 2048
[2026-02-19T09:04:51Z INFO  egui_wgpu] There were 4 available wgpu adapters: {backend: Vulkan, device_type: DiscreteGpu, name: "NVIDIA RTX 6000 Ada Generation", driver: "NVIDIA", driver_info: "580.105.08", vendor: NVIDIA (0x10DE), device: 0x26B1}, {backend: Vulkan, device_type: Cpu, name: "llvmpipe (LLVM 15.0.7, 256 bits)", driver: "llvmpipe", driver_info: "Mesa 23.2.1-1ubuntu3.1~22.04.3 (LLVM 15.0.7)", vendor: Mesa (0x10005)}, {backend: Vulkan, device_type: IntegratedGpu, name: "AMD Unknown (RADV RAPHAEL_MENDOCINO)", driver: "radv", driver_info: "Mesa 23.2.1-1ubuntu3.1~22.04.3", vendor: AMD (0x1002), device: 0x13C0}, {backend: Gl, device_type: Other, name: "NVIDIA RTX 6000 Ada Generation/PCIe/SSE2", driver_info: "3.3.0 NVIDIA 580.105.08", vendor: NVIDIA (0x10DE)}
[ChannelFactory] Initializing with Domain ID: 1, Interface: lo
14:34:53.460740 INFO     Initialize G1_29_ArmController...                           robot_arm.py:70
14:34:53.779716 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:53.880033 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:53.980340 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.080647 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.180962 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.281216 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.381465 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.481801 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.582301 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.682570 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.782966 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.883284 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:54.983598 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112
14:34:55.083906 WARNING  [G1_29_ArmController] Waiting to subscribe dds...          robot_arm.py:112

**Comment by milk163** [comment]
Hello, have you solved this problem? How did you solve it?

**Comment by jayanth100798** [comment]
yes Solved ,
All you need to do is 
clone this repo https://github.com/unitreerobotics/unitree_sim_isaaclab
install requirements and setup using the doc and run this cmd in another terminal along with inference cmd
python sim_main.py --device cuda --task Isaac-PickPlace-Cylinder-G129-Dex1-Joint --enable_dex1_dds --robot_type g129 --enable_cameras 
the above cmd will give the dds topic for the inference script.
change the task and ee based on what policy you trained this - (EX : --task Isaac-PickPlace-Cylinder-G129-Dex1-Joint )

**Comment by milk163** [comment]
Thank you very much.

**Comment by Lee-LEO-H** [comment]
hello,Have you ever encountered this problem when running this python sim_main.py --device cuda --task Isaac-PickPlace-Cylinder-G129-Dex3-Joint --enable_dex3_dds --robot_type g129 --enable_cameras ?

dds_manager: <dds.dds_master.DDSManager object at 0x7090856aba00>
[DDSManager] object 'g129' not found, objects: dict_keys([])
[g1_state] G1 robot DDS communication instance obtained
[DDSManager] object 'dex3' not found, objects: dict_keys([])
[Observations Dex3] DDS communication instance obtained
